react/BarBackButton/index.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+import styles from './styles.styl'
+
+import Icon from '../Icon'
+
+class MaybeLink extends PureComponent {
+  render() {
+    const { children, href, ...restProps } = this.props
+    return href ? (
+      <a href={href} {...restProps}>
+        {children}
+      </a>
+    ) : (
+      <div {...restProps}>{children}</div>
+    )
+  }
+}
+
+MaybeLink.propTypes = {
+  className: PropTypes.string,
+  href: PropTypes.string,
+  onClick: PropTypes.func
+}
+
+export class BarBackButton extends PureComponent {
+  render() {
+    const { disabled, href, onClick } = this.props
+    return (
+      <MaybeLink
+        className={styles['barleft']}
+        href={!disabled && href}
+        onClick={!disabled && onClick}
+      >
+        <Icon icon="back" className={disabled ? 'u-silver' : 'u-coolGrey'} />
+      </MaybeLink>
```

Comment:
It looks like a special icon button to me but it seems not specific to the bar, so I am not comfortable with the naming `BarBackButton`

Reply:
The u-silver, u-coolgrey behavior is specific to BarBackButton. Do you see another instance in Cozy that has this behavior ? I prefer not to abstract a behavior early here. I don't see what we would gain.

Reply:
Maybe BarButton instead of IconButton ?

<BarButton icon='back' />
<BarButton icon='sandwich' />

Reply:
At least `BarButton` is still least specific than `BarBackButton` so 👍

Reply:
So let's go for a `<BackButtion />` I assume that `icon` should now be a prop, or should we wait for the express need ?

Reply:
It should be a prop like in @ptbrowne example